src/csvcubed/utils/qb/cube.py

```diff
@@ -91,3 +92,29 @@ def get_all_units(cube: Cube) -> Set[QbUnit]:
 
     _logger.debug("Discovered units %s", units)
     return units
+
+
+def detect_shape_of_cube(cube: Cube) -> CSVWShape:
+    """
+    Given a cube as input, returns the shape of that cube (Standard or Pivoted)
+    """
+    obs_val_columns = get_columns_of_dsd_type(cube, QbObservationValue)
+
+    all_pivoted = True
+    all_standard_shape = True
+    for obs_val_col in obs_val_columns:
+        all_pivoted = (
+            all_pivoted
+            and obs_val_col.structural_definition.is_pivoted_shape_observation
+        )
+        all_standard_shape = (
+            all_standard_shape
+            and not obs_val_col.structural_definition.is_pivoted_shape_observation
+        )
+
+    if all_pivoted:
+        return CSVWShape.Pivoted
+    elif all_standard_shape:
+        return CSVWShape.Standard
+    else:
+        raise TypeError("The cube cannot be in both standard and pivoted shape")
```

Comment:
Could you make it clear in this error message that it's about some obs val columns being in pivoted shape and others not?